nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/SiteToSiteClient.java

```diff
@@ -107,12 +102,9 @@
      *
      * @return a Transaction to use for sending or receiving data, or
      * <code>null</code> if all nodes are penalized.
-     * @throws org.apache.nifi.remote.exception.HandshakeException he
-     * @throws org.apache.nifi.remote.exception.PortNotRunningException pnre
-     * @throws IOException ioe
-     * @throws org.apache.nifi.remote.exception.UnknownPortException upe
+     * @throws IOException if unable to determine the identifier of the port
      */
-    Transaction createTransaction(TransferDirection direction) throws HandshakeException, PortNotRunningException, ProtocolException, UnknownPortException, IOException;
```

Comment:
i dont believe we can or want to dump these just yet.  I see they all subclass IOException so just throwing IOException would do the job so to speak but then again they express a specific type of failure that a caller could elect to handle specifically.  We could change this logic later such that the cause is of a specific type should the called care but that isn't how it is done now.  We should consider such a change as part of some API improvements over time but for now I'd suggest we keep it out of this pr.

Reply:
I agree; will revert this change. I think opening a separate Jira to discuss error handling (and enforcing consistent behavior across the application) is a good idea.